doc/mgmt/Management Framework.md

```diff
@@ -1081,6 +1092,118 @@ go run gnmi_set.go -replace  /openconfig-acl:acl/acl-sets/acl-set[name=MyACL4][t
+It can be overridden by specifiying a valid TLS private key and certificate file paths through the TELEMETRY table.
+
+###### 3.2.2.5.7 Translib linking
+
+gNMI server will statically link with Translib. For each gNMI request, the server
+invokes Translib API which then invokes appropriate App module to process the request.
+Below is the mapping of gRPC operations to Translib APIs:
+
+ gRPC Method | Translib API      | Request data  | Response data
+-------------|-------------------|---------------|---------------
+ GET         | translib.Get      | path          | status, payload
+ SET/UPDATE  | translib.Update   | path, payload | status
+ SET/REPLACE | translib.Replace  | path, payload | status
+ SET/DELETE  | translib.Delete   | path          | status
+ (ANY OC-RPC)| translib.Action   | RPC,payload   | status, payload
+ SUBSCRIBE   | translib.Subscribe| path          | status, payload
+ CAPABILITIES| translib.GetModels|               | models, versions, encodings
+ 
+
+More details about Translib APIs are in section [3.2.2.6](#3_2_2_6-Translib).
+
+
+###### 3.2.2.5.8 Encodings
+
+gNMI Currently only supports JSON_IETF type encoding in the same was as REST server since translib nativly uses this format.
+
+###### 3.2.2.5.9 Payload Validations
+
+gNMI server does not validate request payload for YANG defined RESTCONF APIs.
+Payload will be validated automatically in lower layers when it gets loaded
+into YGOT bindings.
+
+###### 3.2.2.5.10 Concurrency
+
+gNMI server will accept concurrent requests. Translib provides appropriate locking mechanism - parallel reads and sequential writes.
+
+###### 3.2.2.5.11 API Versioning
+
+gNMI server will allow clients to specify API version through a custom gNMI Extension field BundleVersion with extentsion id 700.
+
+    BundleVersion: 1.0.3
+
+Version text should be in **MAJOR.MINOR.PATCH** format. gNMI server will extract version text from
+the gNMI request protobuf object and pass it to the Translib API as **ClientVersion** argument.
+Section [3.2.2.6.4.2](#322642-version-checks) explains Translib version checking logic.
+
+Version checks are bypassed if Accept-Version header is not present in the request.
+
+For YANG defined RESTCONF APIs, the server's version can be discovered through the standard Capabilities gNMI RPC.
+
+###### 3.2.2.5.12 gNOI RPCs
+
+Along with gNMI RPCS (GET,SET,SUBSCRIBE,CAPABILITIES), custom RPCs are exposed on the [gNOI](https://github.com/openconfig/gnoi) server on the same port as gNMI server. 
+
+gNOI Allows the gNMI server to add custom RPC operations. gNOI Provides a number of standardized RPCs, however currently on the the system/Time RPC is available from those RPCs.
+
+OpenConfig and Sonic customer RPCs are also available via gNOI via the sonic_gnoi.proto file in the telemetry repository. This file provides the protobuf definition for calling these RPCs as well as the format of their arguments and their responses.
```

Comment:
Table at section 3.2.2.5.7 says any oc-rpc will be mapped to `translib.Action`. Is it automatic? Will the sonic_gnoi.proto autogenerated from yang?

Reply:
No, there is no auto-generation currently.

Reply:
In that case, can we change "ANY OC-RPC" to "gNOI RPC" in table 3.2.2.5.7? All oc-yang defined RPCs may not have corresponding gNOI RPCs.